sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -20,21 +25,70 @@
     XML;
 
+    private static final ClientLogger LOGGER = new ClientLogger(SerializerEncoding.class);
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final Map<String, SerializerEncoding> SUPPORTED_MIME_TYPES;
+    private static final TreeMap<String, SerializerEncoding> SUPPORTED_SUFFIXES;
+    private static final SerializerEncoding DEFAULT_ENCODING = JSON;
+
+
+    static {
+        // Encodings and suffixes from: https://tools.ietf.org/html/rfc6838
+        final Comparator<String> comparator = Comparator.comparing(Function.identity(), String::compareToIgnoreCase);
+
+        SUPPORTED_MIME_TYPES = new TreeMap<>(comparator);
+        SUPPORTED_MIME_TYPES.put("text/xml", XML);
+        SUPPORTED_MIME_TYPES.put("application/xml", XML);
+        SUPPORTED_MIME_TYPES.put("application/json", JSON);
+
+        SUPPORTED_SUFFIXES = new TreeMap<>(comparator);
+        SUPPORTED_SUFFIXES.put("xml", XML);
+        SUPPORTED_SUFFIXES.put("json", JSON);
+    }
+
     /**
      * Determines the serializer encoding to use based on the Content-Type header.
      *
-     * @param headers the headers to check the encoding for
-     * @return the serializer encoding to use for the body
+     * @param headers the headers to check the encoding for.
+     * @return the serializer encoding to use for the body. {@link #JSON} if there is no Content-Type header or an
+     * unrecognized Content-Type encoding is returned.
      */
     public static SerializerEncoding fromHeaders(HttpHeaders headers) {
-        String mimeContentType = headers.getValue("Content-Type");
-        if (mimeContentType != null) {
-            String[] parts = mimeContentType.split(";");
-            if (parts[0].equalsIgnoreCase("application/xml") || parts[0].equalsIgnoreCase("text/xml")) {
-                return XML;
-            }
+        final String mimeContentType = headers.getValue(CONTENT_TYPE);
+        if (mimeContentType == null || mimeContentType.isEmpty()) {
+            LOGGER.warning("'{}' not found. Returning default encoding: {}", CONTENT_TYPE, DEFAULT_ENCODING);
+            return DEFAULT_ENCODING;
         }
 
-        return JSON;
+        final SerializerEncoding encoding = SUPPORTED_MIME_TYPES.get(mimeContentType);
+        if (encoding != null) {
+            return encoding;
+        }
+
+        final String[] parts = mimeContentType.split(";");
+        final String[] mimeTypeParts = parts[0].split("/");
+        if (mimeTypeParts.length != 2) {
+            LOGGER.warning("Content-Type '{}' does not match mime-type formatting 'type'/'subtype'. "
+                + "Returning default: {}", parts[0], DEFAULT_ENCODING);
+            return DEFAULT_ENCODING;
+        }
+
+        // Check the suffix if it does not match the full types.
+        final String subtype = mimeTypeParts[1];
+        final int lastIndex = subtype.lastIndexOf("+");
+        if (lastIndex == -1) {
+            return DEFAULT_ENCODING;
+        }
```

Comment:
This could be slightly wrong if we run across something like `text/json`, instead we should check:

```java
final String subType = mimeTypeParts[1];
final int lastIndex = subType.lastIndexOf("+");

final String mimeType = (lastIndex == -1) ? subtype : subtype.substring(lastIndex + 1);
```

Reply:
text/json isn't a well-supported mimetype though, nor is it a proper suffix.